endstate_correction/neq.py

```diff
@@ -49,28 +49,36 @@ def perform_switching(
     ws = []
     # list for all endstate samples (can be empty if saving is not needed)
     endstate_samples = []
-    # list for all switching trajectories (can be empty if saving is not needed)
-    all_switching_trajectories = []
 
     inst_switching = False
     if len(lambdas) == 2:
         print("Instantanious switching: dE will be calculated")
         inst_switching = True
+        if nr_of_switches == -1: # if no specific nr_of_switches is provided (-1 is the default value), use all provided equilibrium samples
+            nr_of_switches = len(samples)
+            print(f"{nr_of_switches} dE values will be calculated")
     elif len(lambdas) < 2:
         raise RuntimeError("increase the number of lambda states")
     else:
         print("NEQ switching: dW will be calculated")
 
     # start with switch
     for switch_index in tqdm(range(nr_of_switches)):
-        # select a random frame
-        random_frame_idx = random.randint(0, len(samples.xyz) - 1)
-        # select the coordinates of the random frame
-        coord = samples.openmm_positions(random_frame_idx)
-        if samples.unitcell_lengths is not None:
-            box_length = samples.openmm_boxes(random_frame_idx)
-        else:
-            box_length = None
+        if inst_switching and nr_of_switches == len(samples): # if all samples should be used for instantanious switching
```

Comment:
Can we add a log or print statement that we are using all samples now?

Reply:
sure!